precompiles/randomness/Randomness.sol

```diff
@@ -1,81 +1,152 @@
 // SPDX-License-Identifier: GPL-3.0-only
 pragma solidity >=0.8.0;
 
+/// @dev Maximum number of random words being requested
+uint32 constant MAX_RANDOM_WORDS = 100;
+/// @dev Minimum number of blocks before a request can be fulfilled
+uint32 constant MIN_DELAY_BLOCKS = 1;
+/// @dev Maximum number of blocks before a request can be fulfilled
+uint32 constant MAX_DELAY_BLOCKS = 10000;
+
 interface Randomness {
-    /// @dev Interface for all randomness consumers
-    ///
-    /// @dev Get relay block number
-    /// @return The relay block number
+    /// @notice The status of the request
+    /// @param Pending The request cannot be fulfilled yet
+    /// @param Ready The request is ready to be fulfilled
+    /// @param Expired The request has expired
+    enum RequestStatus {
+        Pending,
+        Ready,
+        Expired
+    }
+
+    /// @notice The type of randomness source
+    /// @param LocalVRF Randomness VRF using the parachain material as seed
+    /// @param RelayBabeEpoch Randomness VRF using relay material from previous epoch
+    enum RandomnessSource {
+        LocalVRF,
+        RelayBabeEpoch
```

Comment:
This is BabeOneEpochAgo

We are removing BabeTwoEpochs ago so there is no option now that guarantees (relay chain) finality cc @librelois

Reply:
Yes, the finality doesn't provide additional security for us. The Babe consensus uses it as it determine the list of validators that will then be able to produce the next VRF, so it is important to have the additional guarantee of the finality in this case